pkg/operator2/configobservation/oauth/observe_tokenconfig.go

```diff
@@ -0,0 +1,89 @@
+package oauth
+
+import (
+	"k8s.io/klog"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"github.com/openshift/library-go/pkg/operator/configobserver"
+	"github.com/openshift/library-go/pkg/operator/events"
+
+	"github.com/openshift/cluster-authentication-operator/pkg/operator2/configobservation"
+)
+
+const defaultAccessTokenMaxAgeSeconds = float64(86400) // a day
+
+func ObserveTokenConfig(genericlisters configobserver.Listers, recorder events.Recorder, previouslyObservedConfig map[string]interface{}) (map[string]interface{}, []error) {
+	tokenConfigPath := []string{"oauthConfig", "tokenConfig"}
+	listers := genericlisters.(configobservation.Listers)
+	errs := []error{}
+
+	currentTokenConfig, _, err := unstructured.NestedMap(previouslyObservedConfig, tokenConfigPath...)
+	if err != nil {
+		return previouslyObservedConfig, append(errs, err)
+	}
+
+	currentAccessTokenMaxAgeSecondsFloat, _, err := unstructured.NestedFloat64(currentTokenConfig, "accessTokenMaxAgeSeconds")
+	if err != nil {
+		errs = append(errs, err)
+	}
+	currentAccessTokenMaxAgeSeconds := float64(currentAccessTokenMaxAgeSecondsFloat)
```

Comment:
Is this conversion necessary given that `NestedFloat64` returns a float64?

Reply:
I need to re-check this module, I think I left it at a WIP state...